ibis/expr/window.py

```diff
@@ -10,6 +10,24 @@ def _sequence_to_tuple(x):
     return tuple(x) if util.is_iterable(x) else x
 
 
+def _determine_how(preceding):
+    if isinstance(preceding, tuple):
+        start, end = preceding
+        if start is None:
+            offset_type = type(end)
+        else:
+            offset_type = type(start)
+    else:
+        offset_type = type(preceding)
+
+    if issubclass(offset_type, int):
```

Comment:
Can we check `(int, np.integer)` so that passing in a numpy integer works?

Can you also exhaustively test the types that we support? Here you're only checking `int`-like things, and everything else gets a `range` window, but we only know to use range with an interval type. Can you add an `elif` check for interval and raise a `TypeError` in an `else:` block? This way we fail as early as possible on types we don't know how to handle.

Reply:
Good point, fixed/added branches and tests in next commit.